receiver/dockerstatsreceiver/metadata.yaml

```diff
@@ -426,18 +426,36 @@ metrics:
       monotonic: false
   container.memory.total_unevictable:
     enabled: false
-    description: "The amount of memory that cannot be reclaimed. Includes descendant cgroups."
+    description: "The amount of memory that cannot be reclaimed. Includes descendant cgroups (Only available with cgroups v1)."
+    unit: By
+    sum:
+      value_type: int
+      aggregation: cumulative
+      monotonic: false
+  container.memory.anon:
+    enabled: false
+    description: "Amount of memory used in anonymous mappings such as brk(), sbrk(), and mmap(MAP_ANONYMOUS) (Only available with cgroups v2)."
+    extended_documentation: "[More docs](https://www.kernel.org/doc/Documentation/cgroup-v2.txt)"
+    unit: By
+    sum:
+      value_type: int
+      aggregation: cumulative
+      monotonic: false
+  container.memory.file:
+    enabled: true
```

Comment:
@kang-makes can you help me understand which memory metrics someone using cgroups v2 would get with the default config? What's the case for adding `container.memory.file` as a default?

Reply:
cgroups v1 and v2 are different. Really different.

As I stated in the description of this PR, on `v1` the metric `container.memory.total_cache` is activated by default but it is only available for cgroups v1.

As v2 does not have something like `total_cache` that you have to calculate it, from my point of view it makes sense to have all the metrics that are inside `total_cache` in v1, available on `v2` so the user is able to calculate the metric.

The amount of memory consumed by `tmpfs` filesystems are mesudes by `container.memory.file` on v2 and inside `container.memory.total_cache` on v1.

This is the same discussion I was had [here](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/22706#discussion_r1204057802). Can we have the discussion on the same thread?

Reply:
I forgot to ping you @jamesmoessis.

Reply:
Makes sense, thanks for explaining